File: go/easy/1275_find_winner_on_a_tic_tac_toe_game.go
// https://leetcode.com/problems/find-winner-on-a-tic-tac-toe-game
package easy

func Tictactoe(moves [][]int) string {
	n := 3
	rows := make([]int, n)
	cols := make([]int, n)
	diag := 0
	antiDiag := 0
	player := 1

	for _, v := range moves {
		row := v[0]
		col := v[1]

		rows[row] += player
		cols[col] += player

		if row == col {
			diag += player
		}

		if row+col == n-1 {
			antiDiag += player
		}

		if checkTheMove(rows[row], n) || checkTheMove(cols[col], n) ||
			checkTheMove(diag, n) || checkTheMove(antiDiag, n) {
			if player == 1 {
				return "A"
			}
			return "B"
		}

		player *= -1

	}

	if len(moves) == n*n {
		return "Draw"
	}
	return "Pending"
}

func checkTheMove(num, n int) bool {
	if num == n {
		return true
	} else if num == -n {
		return true
	}

	return false
}

File: java/easy/1275_find_winner_on_a_tic_tac_toe_game.java
package java.easy;

class FindWinnerOnATicTacToeGame {
    public String tictactoe(int[][] moves) {
        int n = 3;
        int[] rows = new int[n];
        int[] cols = new int[n];

        int diag = 0;
        int antiDiag = 0;
        int player = 1;

        for (int[] move : moves) {
            int row = move[0];
            int col = move[1];

            rows[row] += player;
            cols[col] += player;

            if (row == col) {
                diag += player;
            }

            if (row + col == n-1) {
                antiDiag += player;
            }

            if (Math.abs(rows[row]) == n || Math.abs(cols[col]) == n ||
                Math.abs(diag) == n || Math.abs(antiDiag) == n) {
                    return player == 1 ? "A" : "B";
            }

            player *= -1;
        }

        if (moves.length + moves[0].length == 9) {
            return "DRAW";
        } else {
            return "PENDING";
        }
    }
}
